src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -163,24 +165,58 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
-        public static unsafe double CopySign(double x, double y)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const double signMask = -0.0;
 
-            long xbits = BitConverter.DoubleToInt64Bits(x);
-            long ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.IsSupported || AdvSimd.IsSupported)
+            {
+                // Create vectors of the elements, and make them float to allow using SSE rather than SSE2 instructions
+                var xvec = Vector128.CreateScalarUnsafe(x).AsSingle();
+                var yvec = Vector128.CreateScalarUnsafe(y).AsSingle();
 
-            // If the sign bits of x and y are not the same,
-            // flip the sign bit of x and return the new value;
-            // otherwise, just return x
+                var mask = Vector128.CreateScalarUnsafe(signMask).AsSingle();
 
-            if ((xbits ^ ybits) < 0)
+                if (Sse41.IsSupported)
+                {
+                    return Sse41.BlendVariable(xvec, yvec, mask).AsDouble().ToScalar();
```

Comment:
`blendvps` is element-wise, so this will overwrite x if y is negative.  There's no bitwise blend (unless you count the new AVX512 ternary as @EgorBo pointed out)

Reply:
ah, right, and I don't think it is easily tweakable

Reply:
fixed